superchain/superchain.go

```diff
@@ -151,7 +152,14 @@ func (c *ChainConfig) setNilHardforkTimestampsToDefault(s *SuperchainConfig) {
 		if overrideValue.IsNil() &&
 			!defaultValue.IsNil() &&
 			reflect.Indirect(defaultValue).Uint() >= *c.SuperchainTime {
-			overrideValue.Set(defaultValue) // use default only if hardfork activated after SuperchainTime
+			// use default only if hardfork activated after SuperchainTime
+			if reflect.Indirect(defaultValue).Uint() > c.Genesis.L2Time {
+				overrideValue.Set(defaultValue)
+			} else {
+				var zero uint64 = 0
+				ptrValue := reflect.ValueOf(&zero)
```

Comment:
could move this above the `for` loop for reuse (and maybe then name the pointer `ptrZero`.